### ISSUE TITLE
"create cluster" sometimes fails after creating the cluster (i/o timeout accessing API endpoint)

### ISSUE THREAD

**Issue by artem-nefedov** [weaveworks/eksctl#2712]
**What happened?**

When running "create cluster", after "cluster" and IAM addon CFN stacks are created, command sometimes fails while trying to access API endpoint. Issue rarely happens. Reproduction rate is ~5-10%.

I never seen this issue with 0.26.0, so I think this might be a regression.

**How to reproduce it?**

eksctl create cluster -f manifest.yaml

manifest.yaml contenst:

```
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig

metadata:
  name: fake-cluster
  region: us-west-2
  tags:
    Environment: fake
    Owner: fake
    Requestor: fake
    Role: fake
  version: "1.17"

iam:
  serviceAccounts:
  - attachPolicy:
      Statement:
      - Action:
        - route53:ChangeResourceRecordSets
        Effect: Allow
        Resource: arn:aws:route53:::hostedzone/*
      - Action:
        - route53:ListHostedZones
        - route53:ListResourceRecordSets
        - route53:ListTagsForResource
        Effect: Allow
        Resource: '*'
      Version: "2012-10-17"
    metadata:
      labels:
        istio-injection: disabled
      name: external-dns
      namespace: external-dns
  - attachPolicy:
      Statement:
      - Action:
        - autoscaling:DescribeAutoScalingGroups
        - autoscaling:DescribeAutoScalingInstances
        - autoscaling:DescribeLaunchConfigurations
        - autoscaling:DescribeTags
        - autoscaling:SetDesiredCapacity
        - autoscaling:TerminateInstanceInAutoScalingGroup
        - ec2:DescribeLaunchTemplateVersions
        Effect: Allow
        Resource: '*'
      Version: "2012-10-17"
    metadata:
      name: cluster-autoscaler
      namespace: kube-system
  withOIDC: true

managedNodeGroups:
- desiredCapacity: 2
  launchTemplate:
    id: lt-12345
  maxSize: 10
  minSize: 2
  name: managed-ng-1
  privateNetworking: true
  tags:
    Environment: fake
    Owner: fake
    Requestor: fake
    Role: fake
secretsEncryption:
  keyARN: FakeKey
```

**Versions**
Please paste in the output of these commands:
```
$ eksctl version
0.29.1
$ kubectl version
v1.18.6
```

**Logs**

```
Creating EKS cluster 'cluster-name'...
[ℹ]  eksctl version 0.29.1
[ℹ]  using region us-west-2
[ℹ]  setting availability zones to [us-west-2b us-west-2a us-west-2d]
[ℹ]  subnets for us-west-2b - public:192.168.0.0/19 private:192.168.96.0/19
[ℹ]  subnets for us-west-2a - public:192.168.32.0/19 private:192.168.128.0/19
[ℹ]  subnets for us-west-2d - public:192.168.64.0/19 private:192.168.160.0/19
[ℹ]  using Kubernetes version 1.17
[ℹ]  creating EKS cluster "cluster-name" in "us-west-2" region with managed nodes
[ℹ]  1 nodegroup (managed-ng-1) was included (based on the include/exclude rules)
[ℹ]  will create a CloudFormation stack for cluster itself and 0 nodegroup stack(s)
[ℹ]  will create a CloudFormation stack for cluster itself and 1 managed nodegroup stack(s)
[ℹ]  if you encounter any issues, check CloudFormation console or try 'eksctl utils describe-stacks --region=us-west-2 --cluster=cluster-name'
[ℹ]  CloudWatch logging will not be enabled for cluster "cluster-name" in "us-west-2"
[ℹ]  you can enable it with 'eksctl utils update-cluster-logging --enable-types={SPECIFY-YOUR-LOG-TYPES-HERE (e.g. all)} --region=us-west-2 --cluster=cluster-name'
[ℹ]  Kubernetes API endpoint access will use default of {publicAccess=true, privateAccess=false} for cluster "cluster-name" in "us-west-2"
[ℹ]  2 sequential tasks: { create cluster control plane "cluster-name", 2 sequential sub-tasks: { 4 sequential sub-tasks: { tag cluster, associate IAM OIDC provider, 3 parallel sub-tasks: { 2 sequential sub-tasks: { create IAM role for serviceaccount "align-system/external-dns", create serviceaccount "align-system/external-dns" }, 2 sequential sub-tasks: { create IAM role for serviceaccount "kube-system/cluster-autoscaler", create serviceaccount "kube-system/cluster-autoscaler" }, 2 sequential sub-tasks: { create IAM role for serviceaccount "kube-system/aws-node", create serviceaccount "kube-system/aws-node" } }, restart daemonset "kube-system/aws-node" }, create managed nodegroup "managed-ng-1" } }
[ℹ]  building cluster stack "eksctl-cluster-name-cluster"
[ℹ]  deploying stack "eksctl-cluster-name-cluster"
[✔]  tagged EKS cluster (<TAGS>)
[ℹ]  building iamserviceaccount stack "eksctl-cluster-name-addon-iamserviceaccount-kube-system-aws-node"
[ℹ]  building iamserviceaccount stack "eksctl-cluster-name-addon-iamserviceaccount-align-system-external-dns"
[ℹ]  building iamserviceaccount stack "eksctl-cluster-name-addon-iamserviceaccount-kube-system-cluster-autoscaler"
[ℹ]  deploying stack "eksctl-cluster-name-addon-iamserviceaccount-kube-system-aws-node"
[ℹ]  deploying stack "eksctl-cluster-name-addon-iamserviceaccount-align-system-external-dns"
[ℹ]  deploying stack "eksctl-cluster-name-addon-iamserviceaccount-kube-system-cluster-autoscaler"
[!]  1 error(s) occurred and cluster hasn't been created properly, you may wish to check CloudFormation console
[ℹ]  to cleanup resources, run 'eksctl delete cluster --region=us-west-2 --name=cluster-name'
[✖]  checking whether namespace "kube-system" exists: Get "https://<ID>.gr7.us-west-2.eks.amazonaws.com/api/v1/namespaces/kube-system": dial tcp <IP-ADDR>:443: i/o timeout
```

**Comment by artem-nefedov** [comment]
Still seeing this with 0.30.0

**Comment by artem-nefedov** [comment]
Fixed in 0.31.0
